### PR TITLE
Fade gallery in after delay to avoid flash

### LINKED ISSUE #13847: [3.0] Product image flickers on load
## EXPLANATION OF THE ISSUE
The product image seems to get resized to a bigger size for a fraction of second when the page loads, which makes the image flicker:

wc-image-load

It looks like this happens only with the image slider option enabled, when visiting a single-image product on Chrome.

## STEPS TO REPRODUCE THE ISSUE
1. Make sure that the product gallery slider feature is enabled
2. Visit a product that has only one image with Chrome

I tested it with the TwentySixteen theme, but it also happens with other themes as well.

### PR DESCRIPTION
**Pull request by mikejolley**
Closes #13847

### CODE REVIEW

**Comment by mikejolley** [comment]
@jameskoster Not sure if you'll like this one - also not sure if 3.0 or 3.1. Thoughts?

**Comment by jameskoster** [comment]
Fade effect: 👍 

As for the execution, I think we should use CSS for the animation. Ideally we should make this theme agnostic as well, IE nothing in woocommerce.css. 

Instead, add the `opacity: 0; transition: whatever;` style inline, in the template, then just use `.css()` to set the opacity after the slider is loaded rather than an arbitrary delay (I believe there's a callback we can use in flexslider). 

I'm not sure how we address browsers with js disabled with this approach though. Do we need to worry about this? Does our checkout even function without js?

**Comment by mikejolley** [comment]
@jameskoster will try this first https://github.com/woocommerce/woocommerce/pull/13863/files

**Comment by jameskoster** [comment]
Discussed on Slack. :shipit:

**Review by jameskoster — APPROVED** [review]

File: assets/js/frontend/single-product.js [hunk 1]
```diff
@@ -108,6 +108,8 @@ jQuery( function( $ ) {
 		if ( this.flexslider_enabled ) {
 			this.initFlexslider();
 			$target.on( 'woocommerce_gallery_reset_slide_position', this.onResetSlidePosition );
+		} else {
+			$( '.woocommerce-product-gallery' ).css( 'opacity', 1 );
```
Comment (franticpsyx) [inline assets/js/frontend/single-product.js]: @mikejolley -- Did you mean to use `$target` here? The object is agnostic to the classes that the gallery container has.

File: assets/js/frontend/single-product.js [hunk 2]
```diff
@@ -136,6 +138,9 @@ jQuery( function( $ ) {
 			animationSpeed: wc_single_product_params.flexslider.animationSpeed,
 			animationLoop:  wc_single_product_params.flexslider.animationLoop, // Breaks photoswipe pagination if true.
 			start: function() {
+				// Fade in.
+				$( '.woocommerce-product-gallery' ).css( 'opacity', 1 );
```
Comment (franticpsyx) [inline assets/js/frontend/single-product.js]: Same as above (`this.$target`).

**Review by manospsyx — COMMENTED** [review]
Added some comments 👍

**Pull request merged**